tests/e2e/config/jest-custom-environment.js

```diff
@@ -0,0 +1,29 @@
+const { TestEnvironment } = require( 'jest-environment-puppeteer' );
+require( 'jest-circus' );
+
+class JestCustomEnvironment extends TestEnvironment {
+	async teardown() {
+		if ( this.global.page ) {
+			// Wait a few seconds before tearing down the page so we
+			// have time to take screenshots and handle other events
+			await this.global.page.waitForTimeout( 1000 );
```

Comment:
Did you have a chance to investigate some alternatives than using a timeout of 1s? 
Just curious, not a blocker - it'll do for now :) albeit it looks like it's a pretty common practice :) https://github.com/search?q=%22Wait+a+few+seconds+before+tearing+down+the+page+so+we%22&type=code

Reply:
I grabbed the solution from [this discussion](https://github.com/argos-ci/jest-puppeteer/issues/131#issuecomment-493267937) since it didn't seem to be built in any of the packages we are using.

However, it looks like the timeout shouldn't be necessary anymore since now jest and jest-circus provide [full support for async event handlers](https://github.com/jestjs/jest/pull/9397) since version 25.3.0.

I'll remove it!